pr_parser.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python
+
+"""
+pr_parser.py: SCRIPTS that analyse pr infomation and write parsed pr infomation
+(especially the multi-relate-pr) to post-build proprerties for downstream job.
+"""
+__author__ = "Alan Wei"
+
+
+import argparse
+import sys
+import os
+import github
+
+from github import Github
+
+class PrParser(object):
+
+    def __init__(self):
+        """
+        __repo - RackHD/on-xxxx, The repo name of which repo triggers the build 
+        __target_branch - master etc., The target branch associated with pr
+        __merge_commit_sha - origin/pr/1/merge, The merge_commit_sha associated with the pr
```

Comment:
It's official name is "merge_commit_sha", when a pr is mergable(no conflicts with target branch), github will create a "hidden" branch which has merged pr to target branch. We can checkout this branch with "origin/pr/1/merge" or a sha1 code like "2d8bb3cec1c529dd80b3d014f6c12c69d482ba42", they are the same to git.